api/base/exceptions.py

```diff
@@ -80,20 +88,88 @@ class Conflict(APIException):
     default_detail = ('Resource identifier does not match server endpoint.')
 
 
-class InvalidQueryStringError(JSONAPIException):
+class JSONAPIParameterException(JSONAPIException):
+    def __init__(self, detail=None, parameter=None):
+        source = {
+            'parameter': parameter
+        }
+        super(JSONAPIParameterException, self).__init__(detail=detail, source=source)
+
+
+class JSONAPIAttributeException(JSONAPIException):
+    def __init__(self, detail=None, attribute=None):
+        source = {
+            'pointer': '/data/attributes/{}'.format(attribute)
+        }
+        super(JSONAPIAttributeException, self).__init__(detail=detail, source=source)
+
+
+class InvalidQueryStringError(JSONAPIParameterException):
     """Raised when client passes an invalid value to a query string parameter."""
     default_detail = 'Query string contains an invalid value.'
     status_code = http.BAD_REQUEST
 
-    def __init__(self, detail=None, parameter=None):
-        super(InvalidQueryStringError, self).__init__(detail=detail, source={
-            'parameter': parameter
-        })
 
+class InvalidFilterOperator(JSONAPIParameterException):
+    """Raised when client passes an invalid operator to a query param filter."""
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None, value=None, valid_operators=None):
+        if value and not detail:
+            valid_operators = valid_operators or ['eq', 'lt', 'lte', 'gt', 'gte', 'contains', 'icontains']
+            detail = "Value '{0}' is not a supported filter operator; use one of {1}.".format(
+                value,
+                valid_operators
```

Comment:
For readability, it'd be good to make this a comma-separated list (i.e. with str.join). That will also ensure that sets and tuples are displayed correctly.